query_builders/query_builder.py

```diff
@@ -311,11 +313,21 @@ def list_where_rel_conditions_per_dict(mp: dict) -> ([], {}):
                 leftdir = (check[-1] if check[-1] in ['<'] else '')
                 rightdir = (check[-1] if check[-1] in ['>'] else '')
                 inner_cond_list = []
+                matched_lst=[]
                 assert isinstance(content, dict)
                 for operator_, lst in content.items():
                     assert operator_ in ['include', 'exclude', 'include_matched', 'exclude_matched']
-                    lst_matched_str = "[" + ", ".join(['`' + item + '`' for item in lst]) + "]"
-                    lst_str = "(" + " OR ".join(['x:`' + item + '`' for item in lst]) + ")"
+                    for item in lst:
+                        if re.search(r"\b" + re.escape('uri') + r"\b", item):
+                            uri = ast.literal_eval("{"+item+"}")
+                            for lst, val in uri.items():
+                                for key, value in val.items():
+                                    matched_lst.append(lst)
+                                    inner_cond_list.append(f"x.{key} in {value}")
```

Comment:
I don't think it will work as expected. It is more complicated than that.

To explain better what is expected here is a more complex example test:
```
test_map = {
        'nobs': {
            'EXISTS>': {'include': ["'Test' : {'uri': ['neo4j://graph.schema#Term/S980028/S91301']}"]},
            'NOT EXISTS': {'exclude': [{'Ser': {'rdfs:label': ['Y']}}, 'Pop', 'Asta']}
        }
    }
#then
expected_cypher_list = [
        "EXISTS {MATCH (`nobs`)-[]->(x) WHERE (x.uri in ['neo4j://graph.schema#Term/S980028/S91301'] AND x:`Test`)}",
        'NOT EXISTS {MATCH (`nobs`)-[]-(x) WHERE NOT ((x.`rdfs:label` in ['Y'] AND x:`Ser`) OR x:`Pop` OR x:`Asta`)}'
    ]
```
i.e. the propery value check goes along with the label check combined by AND, the combined conditions are separated with OR


another example with 'exclude_matched' (this will be more similar to the one you implemented):
```
test_map = {
        'nobs': {
            'EXISTS>': {'include': ["'Test' : {'uri': ['neo4j://graph.schema#Term/S980028/S91301']}"]},
            'NOT EXISTS': {'exclude**_matched**': [{'Ser': {'rdfs:label': ['Y']}}, 'Pop', 'Asta']}
        }
    }
#then
expected_cypher_list = [
        "EXISTS {MATCH (`nobs`)-[]->(x) WHERE (x.uri in ['neo4j://graph.schema#Term/S980028/S91301'] AND x:`Test`)}",
        'NOT EXISTS {MATCH (`nobs`)-[]-(x) WHERE NOT NOT (x in [`Ser`, `Pop`, `Asta`] AND `Ser`.`rdfs:label` in ['Y'])
    ]
```

We actually don't need the the cases for 'include_matched' and 'exclude_matched' right now,
so if it is too difficult, in the first instance you may leave these unchanged.
Just add assert statements that the items of the inner list are strings only (no dicts allowed in case operator in ['include_matched', 'exclude_matched'])
"NOT EXISTS {MATCH (`nobs`)<-[]-(x) WHERE NOT (x in [`Ser`, `Pop`, `Asta`])}"